includes/models/class-fee-plan.php

```diff
@@ -269,6 +269,21 @@ public function calculate_expiration_time( $base_time = null ) {
         return date( 'Y-m-d H:i:s', $expire_time );
     }
 
+	/**
+	 * Checks if the current plan is premium and enabled.
+	 * This checks if payments are turned on and the price is set.
+	 *
+	 * @since x.x
+	 *
+	 * @return bool
+	 */
+	public function is_enabled_premium() {
+		if ( ( $this->amount > 0.0 ) && ! wpbdp_payments_possible() ) {
```

Comment:
This looks like it will end up creating a loop since we need to check plans in order to now if payments are on.

It looks like this function should be removed completely, unless it is used to check if the plan has a price:
```
public function is_paid_plan() {
return $this->amount > 0.0;
}
```

Reply:
Agreed :)